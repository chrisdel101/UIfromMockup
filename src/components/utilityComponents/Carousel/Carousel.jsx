import React from 'react'
import { Carousel } from 'react-bootstrap'
import Image from '../Image/Image'
import Review from '../Review/Review'
import './index.css'

function renderReviews(props) {
  return props.reviews.map((review, i) => {
    return (
      <Carousel.Item key={i}>
        <Review
          src={review.src}
          name={review.name}
          text={review.text}
          key={i}
        />
      </Carousel.Item>
    )
  })
}
function renderWorks(props) {
  return props.works.map((review, i) => {
    return (
      <Carousel.Item key={i}>
        <Image src={review.src} />
      </Carousel.Item>
    )
  })
}

function myCarousel(props) {
  if (!props.instance) return
  return (
    <Carousel interval={30000} className="work">
      {props.instance === 'reviews' ? renderReviews(props) : renderWorks(props)}
    </Carousel>
  )
}

export default myCarousel
