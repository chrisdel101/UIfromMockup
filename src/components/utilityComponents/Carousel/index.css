.carousel {
  height: 200px;
  max-width: 800px;
  margin: auto;
}
.carousel-control-next-icon,
.carousel-control-prev-icon {
  filter: opacity(0.5) drop-shadow(0 0 0 black);
}
