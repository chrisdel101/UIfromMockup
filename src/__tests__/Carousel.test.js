import React from 'react'
import Carousel from '../components/utilityComponents/Carousel/Carousel.jsx'
import renderer from 'react-test-renderer'

test('create and match snapshot', () => {
  const reviews = [
    {
      icon: 'icon',
      src: 'http://place-puppy.com/101x101',
      name: 'Joe',
      text:
        'Justo voluptua gubergren duo eos sadipscing amet et rebum dolores ut, ut ea sadipscing et tempor lorem lorem diam, dolore.'
    },
    {
      icon: 'icon',
      src: 'http://place-puppy.com/101x102',
      name: 'Sam',
      text:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Adipisci, nihil nisi. Molestias ab facilis esse voluptatibus vero excepturi optio laborum, rem, modi nihil expedita. Omnis necessitatibus dignissimos qui minus mollitia?'
    }
  ]
  const component = renderer.create(
    <Carousel instance="reviews" reviews={reviews} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
