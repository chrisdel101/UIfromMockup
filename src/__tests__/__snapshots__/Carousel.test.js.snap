// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create and match snapshot 1`] = `
<div
  className="work carousel slide"
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      onClick={[Function]}
    />
     
    <li
      className={null}
      onClick={[Function]}
    />
     
  </ol>
  <div
    className="carousel-inner"
  >
    <div
      className="carousel-item active carousel-item"
    >
      <div
        className="review"
      >
        <div
          className="image-container"
          id=""
        >
          <img
            src="http://place-puppy.com/101x101"
          />
        </div>
        <div
          className="review-text"
        >
          <div
            className="info-text"
          >
            <p>
              "Justo voluptua gubergren duo eos sadipscing amet et rebum dolores ut, ut ea sadipscing et tempor lorem lorem diam, dolore."
            </p>
          </div>
          <div
            className="review-name-conatiner"
          >
            <div
              className="dash"
            />
            <span
              className="review-name"
            >
               
              Joe
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="carousel-item carousel-item"
    >
      <div
        className="review"
      >
        <div
          className="image-container"
          id=""
        >
          <img
            src="http://place-puppy.com/101x102"
          />
        </div>
        <div
          className="review-text"
        >
          <div
            className="info-text"
          >
            <p>
              "Lorem ipsum dolor sit amet consectetur adipisicing elit. Adipisci, nihil nisi. Molestias ab facilis esse voluptatibus vero excepturi optio laborum, rem, modi nihil expedita. Omnis necessitatibus dignissimos qui minus mollitia?"
            </p>
          </div>
          <div
            className="review-name-conatiner"
          >
            <div
              className="dash"
            />
            <span
              className="review-name"
            >
               
              Sam
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <a
    className="carousel-control-prev"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control-next"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
